### PR TITLE
XBlock tutorial overview

### PR DESCRIPTION
**Pull request by Lyla-Fischer**
A significant rewrite of the overview of the XBlock tutorial, in order to clarify the difference between developing XBlocks and developing for edx-platform.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Lyla-Fischer! I've created [OSPR-914](https://openedx.atlassian.net/browse/OSPR-914) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 1]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
+
+In educational applications, XBlocks can be used to represent individual 
+problems, web-formatted text and videos, interactive simulations and labs, or 
+collaborative learning experiences. Furthermore, XBlocks are composable, 
+allowing an XBlock developer to control the display of other XBlocks to compose
+lessons, sections, and entire courses.
 
 *****************************
-XBlock API and Runtimes
+XBlock Specification and Runtimes
```
Comment (Lyla-Fischer) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: @nedbat Throughout this overview, I tried referring to the XBlock API as a specification so that nobody confuses the API with a RESTful API. Opinions?

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 2]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
 
-Developers can extend the types of content available in the edX Platform by
-creating and deploying XBlocks. EdX provides the `XBlock SDK`_ to support the
-creation of new XBlocks.
+XBlocks are like miniature web applications: they maintain state in a storage
+layer, render themselves through views, and process user actions through
+handlers. XBlocks differ from web applications in that they render only a small piece of
+a complete web page. Like HTML ``<div>`` tags, XBlocks can represent components as small as a
+paragraph of text, a video, or a multiple choice input field, or as large as a
+section, a chapter, or an entire course.
```
Comment (Lyla-Fischer) [inline en_us/xblock-tutorial/source/overview/introduction.rst:81]: The intro paragraph for developers is taken from what used to be a dedicated section "XBlocks compared to web applications" that was hidden at the bottom of the page.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 3]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
-of content. The XBlock API and XBlock SDK are available for developers to
-create new types of XBlocks for course teams to use.
-
 ***********************
 XBlocks for Developers
 ***********************
 
-Developers can extend the types of content available in the edX Platform by
-creating and deploying XBlocks. EdX provides the `XBlock SDK`_ to support the
-creation of new XBlocks.
```
Comment (Lyla-Fischer) [inline en_us/xblock-tutorial/source/overview/introduction.rst:69]: I removed this explanation so that developers don't think that they can use XBlocks to extend edx-platform outside of the specification provided by XBlocks.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 4]
```diff
@@ -89,13 +94,14 @@ This tutorial is meant to guide developers through the process of creating an
...
 
+EdX provides the `XBlock SDK`_ to support the creation of new XBlocks. 
 Developers should also see the :ref:`xblockapi:EdX XBlock API Guide`.
 
 =========================================
 XBlock Independence and Interoperability
 =========================================
 
-You must design your XBlock to meet two goals.
+You must design your XBlock to meet two criteria.
```
Comment (Lyla-Fischer) [inline en_us/xblock-tutorial/source/overview/introduction.rst:111]: must seems like it should reference criteria. Goals seems like a "should" word to me.

**Comment by Lyla-Fischer** [comment]
@nedbat @cpennington I was trying to put off working on tutorial-style documentation, but recent emails on edx-code made me think that this clarification was nessessary, at least in the overview. Can you check over this commit to make sure that I am representing XBlocks accurately?

**Comment by sarina** [comment]
@Lyla-Fischer FYI, we have a whole open source pull request process so teams can planfully attend to work that would otherwise be interrupt-driven. Generally, we ask that you are free to ask questions on PRs but avoid pinging individual developers, as team assignments and availability may be different from when you were here. For example, Cale is now on the Destination team and likely doesn't have time to address XBlock issues currently.

The OSPR process is designed to make sure PRs are routed correctly to the current domain experts or owners for the product.

**Comment by sarina** [comment]
Obviously my above comment is completely moot if you're actively working with specific people, of course. I want to be flexible, but also make sure devs don't think they need to drop everything to jump into a PR when they don't have to.

**Comment by Lyla-Fischer** [comment]
Oh, cool. I didn't realize that the OSPR was as much about protecting dev time as it was about being an advocate for the open source community with boots in the office. I don't know how much Cale is going to be involved with XBlock stuff, but he did a lot of the development in the area, and I think that it makes sense for him to take a look at overview documentation if he is going to be involved in it again. If he has completely moved on to other things, that is fine, but from a community management perspective, and thinking of him as "just part of the community", I think that it makes a lot of sense for him to be involved. I want to be sure that we don't start making progress while excluding Cale from something that he contributed a lot of time and energy to.

**Comment by sarina** [comment]
On the flip side, as new developers join the org, they can make decisions and help shape the future of the product that they're responsible for.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 5]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
-
-A primary advantage to XBlocks is that they are deployable. The code you write
-can be deployed in any instance of the edX Platform or other XBlock runtime
-application, then used by any course team using that system.
-
-By combining XBlocks from a wide variety of sources, from text and video, to
-multiple choice and numerical questions, to sophisticated collaborative and
-interactive learning laboratories, course teams can create rich and engaging
-courseware.
+The XBlock specification is a component architecture developped by edX to
```
Comment (cpennington) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: typo: `developped`

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 6]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
-As described below, the edX Platform uses the XBlock API, and is the most
-common XBlock runtime.
-
-If you are planning to use your XBlock in a different rutime, check the
-documentation to understand the runtime's requirements.
+A web-app can turn itself into an :ref:`XBlock runtime<XBlock Runtimes>` by 
+implementing the XBlock specification. XBlock runtimes can compose web 
+pages out of XBlocks that were developped by programmers who do not need 
+to know anything about the other components a web page might be using or 
+displaying.
```
Comment (Lyla-Fischer) [inline en_us/xblock-tutorial/source/overview/introduction.rst:44]: This was something that I heard at one point in time, but I am wondering if that was an aspirational statement about what XBlocks should be, rather than a statement about what they currently are. IIRC, I have heard that XBlock developers still need to consider how their XBlocks interact with other XBlocks. It's part of the "Criteria" section in this doc, which I preserved, but couldn't find much guidance around what kinds of interactions XBlocks can have with each other. 

Tagging the maintainers because I am so unsure about the current facts of the repository, and don't want to document inaccurately. @nedbat @cpennington

File: en_us/xblock-tutorial/source/overview/introduction.rst (hunk 1 above)
Comment (nedbat) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: "API" is not limited to RESTful http-based APIs.  The P stands for Programming, we shouldn't cede this territory to the network kiddies.  Programming interfaces form the contract between the XBlock runtime and the XBlock that our readers are creating.  It's fine (and even important) to call it an API.   Feel free to add text explaining what we mean by API here.

File: en_us/xblock-tutorial/source/overview/introduction.rst (hunk 3 above)
Comment (nedbat) [inline en_us/xblock-tutorial/source/overview/introduction.rst:69]: I see what you mean about the word "extend."  I think it is useful to point out that developing an XBlock is a way to add a new kind of content to Open edX.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 7]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
-can be deployed in any instance of the edX Platform or other XBlock runtime
-application, then used by any course team using that system.
-
-By combining XBlocks from a wide variety of sources, from text and video, to
-multiple choice and numerical questions, to sophisticated collaborative and
-interactive learning laboratories, course teams can create rich and engaging
-courseware.
+The XBlock specification is a component architecture developped by edX to 
+make it easier to create new online educational experiences. While it was 
+created by an organization with a focus in education, the technology has applications
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: "While it was created..." is hard to parse: will the next words be, "the designers were eating ice cream"?  How about, "XBlock was designed by edX with a focus in education, but the technology..."

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 8]
```diff
@@ -14,59 +14,64 @@ This section introduces XBlocks.
...
+for any web app that needs to use multiple independent components that are displayed on 
+a single web page. 
+
+From edX's perspective, an XBlock developer
```
Comment (nedbat) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: I'm not sure what "From edX's perspective" adds here.

File: en_us/xblock-tutorial/source/overview/introduction.rst (hunk 6 above)
Comment (nedbat) [inline en_us/xblock-tutorial/source/overview/introduction.rst:44]: "A web-app can turn itself into" is a little twisty, and imbues the app with agency.  How about, "Any web application can be an XBlock runtime by implementing..."

Also, "developped" again.

**Comment by Lyla-Fischer** [comment]
Thanks for the comments, @nedbat. 

@mhoeber any opinions on formatting or copy editing?

**Comment by sarina** [comment]
@Lyla-Fischer Mark Hoeber is no longer with edX. I can schedule a review with our documentation team.

**Comment by sarina** [comment]
@Lyla-Fischer continuing our above discussion, that's another reason to have the process - as team assignments and members change, people within edX are in the best position to understand who needs to do review. Before Mark left, he had not been working on Open edX stuff for about 6 months.

**Comment by Lyla-Fischer** [comment]
@sarina cool. I'm glad you're around then. 

For context: I had a couple of email exchanges with Mark re: XBlock documentation. He did not tell me that he had left, but it made sense on my end to continue the conversation without seeking a re-introduction.

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 9]
```diff
@@ -14,59 +14,71 @@ This section introduces XBlocks.
...
-courseware.
+The XBlock specification is a component architecture designed to 
+make it easier to create new online educational experiences. 
+XBlock was developed by edX, which has a focus in education, but the technology 
+can be used in web applications that need to use multiple independent 
+components and display those components on a single web page. 
+
+An XBlock developer 
+does not need to download and run the entire edx-platform developer stack 
+or to know anything about the technolgies that edX uses provide the XBlock
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: the technolgies that edX uses provide --> the technologies that edX uses ^to provide

File: en_us/xblock-tutorial/source/overview/introduction.rst [hunk 10]
```diff
@@ -14,59 +14,71 @@ This section introduces XBlocks.
...
-create new types of XBlocks for course teams to use.
-
 ***********************
 XBlocks for Developers
 ***********************
 
-Developers can extend the types of content available in the edX Platform by
-creating and deploying XBlocks. EdX provides the `XBlock SDK`_ to support the
-creation of new XBlocks.
+Developers can select from functionality developed by the openedX community by
```
Comment (lamagnifica) [inline en_us/xblock-tutorial/source/overview/introduction.rst]: openedX --> Open edX

**Comment by lamagnifica** [comment]
a few nits, :+1: 
thanks @Lyla-Fischer !

**Comment by Lyla-Fischer** [comment]
@lamagnifica @nedbat @sarina ping?

**Comment by nedbat** [comment]
@Lyla-Fischer sorry for the delay.  I'm working through your requests in parallel with other work.  I expect to get to this this week.

**Comment by Lyla-Fischer** [comment]
No problem. I just wanted to make sure it didn't get forgotten. I keep forgetting about it because it is not in a repository that I check very often, and I wanted to make sure you weren't in the same situation.
